src/components/SidePane/Help/index.tsx

```diff
@@ -1,25 +1,84 @@
+// @ts-expect-error This is a workaround to display HelpPane on top of everything,
+// Modal from react-native can't be used here, as it would block interactions with the rest of the app
+import ModalPortal from 'react-native-web/dist/exports/Modal/ModalPortal';
+import FocusTrapForModal from '@components/FocusTrap/FocusTrapForModal';
+import SidePaneOverlay from '@components/SidePane/SidePaneOverlay';
 import useKeyboardShortcut from '@hooks/useKeyboardShortcut';
 import useResponsiveLayout from '@hooks/useResponsiveLayout';
+import useRootNavigationState from '@hooks/useRootNavigationState';
 import useSafeAreaPaddings from '@hooks/useSafeAreaPaddings';
 import useThemeStyles from '@hooks/useThemeStyles';
 import CONST from '@src/CONST';
+import NAVIGATORS from '@src/NAVIGATORS';
+import ONYXKEYS from '@src/ONYXKEYS';
 import HelpContent from './HelpContent';
 import type HelpProps from './types';
 
-function Help({sidePaneTranslateX, closeSidePane}: HelpProps) {
+function Help({sidePaneTranslateX, closeSidePane, shouldHideSidePaneBackdrop}: HelpProps) {
     const styles = useThemeStyles();
     const {isExtraLargeScreenWidth, shouldUseNarrowLayout} = useResponsiveLayout();
     const {paddingTop, paddingBottom} = useSafeAreaPaddings();
 
-    useKeyboardShortcut(CONST.KEYBOARD_SHORTCUTS.ESCAPE, () => closeSidePane(), {isActive: !isExtraLargeScreenWidth});
+    const isInRHP = useRootNavigationState((state) => state?.routes.at(-1)?.name === NAVIGATORS.RIGHT_MODAL_NAVIGATOR);
+    const [modal] = useOnyx(ONYXKEYS.MODAL);
+    const isInNarrowPaneModal = !!modal?.isVisible || isInRHP;
+
+    const onCloseSidePaneOnSmallScreens = () => {
+        if (isExtraLargeScreenWidth) {
+            return;
+        }
+
+        closeSidePane();
+    };
+
+    // Close side pane on escape key press
+    useKeyboardShortcut(CONST.KEYBOARD_SHORTCUTS.ESCAPE, () => closeSidePane(), {isActive: !isExtraLargeScreenWidth, shouldBubble: false});
+
+    // Close side pane on small screens when navigation keyboard shortcuts are used
+    useKeyboardShortcut(CONST.KEYBOARD_SHORTCUTS.SEARCH, onCloseSidePaneOnSmallScreens, {shouldBubble: true});
+    useKeyboardShortcut(CONST.KEYBOARD_SHORTCUTS.NEW_CHAT, onCloseSidePaneOnSmallScreens, {shouldBubble: true});
+    useKeyboardShortcut(CONST.KEYBOARD_SHORTCUTS.SHORTCUTS, onCloseSidePaneOnSmallScreens, {shouldBubble: true});
+
+    // Web back button: push history state and close side pane on popstate
+    useEffect(() => {
+        window.history.pushState({isSidePaneOpen: true}, '', null);
+        const handlePopState = () => {
+            if (isExtraLargeScreenWidth) {
+                return;
+            }
+
+            closeSidePane();
+        };
+
+        window.addEventListener('popstate', handlePopState);
+        return () => window.removeEventListener('popstate', handlePopState);
+        // eslint-disable-next-line react-compiler/react-compiler, react-hooks/exhaustive-deps
+    }, []);
```

Comment:
This part handles the back button behavior on the web. We use a similar approach for Modals/Popovers. It's not perfect, but it works in most cases:

https://github.com/user-attachments/assets/9fccfdbe-b051-4ad6-ad9a-da03ab913355

One known bug that I won't be able to fix is that when opening both a Modal (`src/components/Modal/index.tsx`) and the side pane, pressing the back button dismisses both at once instead of handling them separately:

https://github.com/user-attachments/assets/03102bc9-45ab-418b-87f9-fe3517a723e1